Automattic/wp-calypso
# media: replace upload thunks with mutations

#### Changes proposed in this Pull Request

* Move any thunks related to uploading media items to use React Query, more specifically
  * Dropzone (both `/media` and the media library modal)
  * Add Media button (both `/media` and the media library modal)
  * Add via URL (both `/media` and the media library modal)
  * Add external media (e.g. Pexels; both `/media` and the media library modal)
  * Upload site icon (at `/settings/general/:site`)
  * Podcast cover image (at `/settings/podcasting/:site`)

Please note, instead of opening several PRs I splitted relevant changes according to their upload function. I think it makes sense to review those changes commit by commit.

#### Testing instructions

* Go to each section mentioned above and verify that uploading one or several items still works exactly as on prod
* Make sure to also test the error case by blocking the network request accordingly

Related to #58574
